src/Plugin/search_api/processor/DgiImageDiscovery.php

```diff
@@ -0,0 +1,125 @@
+ *   stages = {
+ *     "add_properties" = 0,
+ *   },
+ *   locked = true,
+ *   hidden = true,
+ * )
+ */
+class DgiImageDiscovery extends ProcessorPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The DGI Image Discovery service.
+   *
+   * @var \Drupal\dgi_image_discovery\ImageDiscoveryInterface
+   */
+  protected $imageDiscovery;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    ImageDiscoveryInterface $image_discovery,
+    EntityTypeManagerInterface $entity_type_manager
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->imageDiscovery = $image_discovery;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('dgi_image_discovery.service'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+
+    if (!$datasource) {
+      $definition = [
+        'label' => $this->t('DGI Image Discovery'),
+        'description' => $this->t('Styled Image Url which can then be passed to the image src.'),
+        'type' => 'string',
+        'is_list' => FALSE,
+        'processor_id' => $this->getPluginId(),
+      ];
+      $properties['dgi_image_discovery'] = new DgiImageDiscoveryProperty($definition);
+    }
+
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addFieldValues(ItemInterface $item) {
+    $entity = $item->getOriginalObject()->getValue();
+    $value = NULL;
+
+    // Get the image discovery URL.
+    if (!$entity->isNew() && $entity instanceof NodeInterface) {
+      $event = $this->imageDiscovery->getImage($entity);
+      $media = $event->getMedia();
+      if (!empty($media)) {
+        $media_source = $media->getSource();
+        $file_id = $media_source->getSourceFieldValue($media);
+        $image = $this->entityTypeManager->getStorage('file')->load($file_id);
+      }
+
+      $fields = $item->getFields(FALSE);
+      $fields = $this->getFieldsHelper()->filterForPropertyPath($fields, NULL, 'dgi_image_discovery');
+      foreach ($fields as $field) {
+        $config = $field->getConfiguration();
+        $image_style = $config['image_style'];
+        if (!empty($image)) {
+          $value = $this->entityTypeManager->getStorage('image_style')->load($image_style)
+            ->buildUrl($image->getFileUri());
+        }
```

Comment:
Could probably move around some logic to improve readability here:

```suggestion
      if (empty($media)) {
        return;
      }
      
      $media_source = $media->getSource();
      $file_id = $media_source->getSourceFieldValue($media);
      $image = $this->entityTypeManager->getStorage('file')->load($file_id);
      
      if (empty($image)) {
        return;
      }

      $fields = $item->getFields(FALSE);
      $fields = $this->getFieldsHelper()->filterForPropertyPath($fields, NULL, 'dgi_image_discovery');
      foreach ($fields as $field) {
        $config = $field->getConfiguration();
        $image_style = $config['image_style'];
        $value = $this->entityTypeManager->getStorage('image_style')->load($image_style)
            ->buildUrl($image->getFileUri());
```